lib/codecs/src/decoding/framing/chunked_gelf.rs

```diff
@@ -0,0 +1,715 @@
+use crate::{BytesDecoder, StreamDecodingError};
+
+use super::{BoxedFramingError, FramingError};
+use bytes::{Buf, Bytes, BytesMut};
+use derivative::Derivative;
+use snafu::{ensure, Snafu};
+use std::any::Any;
+use std::collections::HashMap;
+use std::sync::{Arc, Mutex};
+use std::time::Duration;
+use tokio;
+use tokio::task::JoinHandle;
+use tokio_util::codec::Decoder;
+use tracing::{info, warn};
+use vector_config::configurable_component;
+
+const GELF_MAGIC: [u8; 2] = [0x1e, 0x0f];
+const GELF_MAX_TOTAL_CHUNKS: u8 = 128;
+const DEFAULT_TIMEOUT_MILLIS: u64 = 5000;
+const DEFAULT_PENDING_MESSAGES_LIMIT: usize = 1000;
+
+const fn default_timeout_millis() -> u64 {
+    DEFAULT_TIMEOUT_MILLIS
+}
+
+const fn default_pending_messages_limit() -> usize {
+    DEFAULT_PENDING_MESSAGES_LIMIT
+}
+
+/// Config used to build a `ChunkedGelfDecoder`.
+#[configurable_component]
+#[derive(Debug, Clone, Default, PartialEq, Eq)]
+pub struct ChunkedGelfDecoderConfig {
+    /// Options for the chunked GELF decoder.
+    #[serde(default, skip_serializing_if = "vector_core::serde::is_default")]
+    pub chunked_gelf: ChunkedGelfDecoderOptions,
+}
+
+impl ChunkedGelfDecoderConfig {
+    /// Build the `ChunkedGelfDecoder` from this configuration.
+    pub fn build(&self) -> ChunkedGelfDecoder {
+        ChunkedGelfDecoder::new(
+            self.chunked_gelf.timeout_millis,
+            self.chunked_gelf.pending_messages_limit,
+        )
+    }
+}
+
+/// Options for building a `ChunkedGelfDecoder`.
+#[configurable_component]
+#[derive(Clone, Debug, Derivative, PartialEq, Eq)]
+pub struct ChunkedGelfDecoderOptions {
+    /// The timeout, in milliseconds, for a message to be fully received. If the timeout is reached, the
+    /// decoder drops all the received chunks of the incomplete message and starts over.
+    #[serde(
+        default = "default_timeout_millis",
+        skip_serializing_if = "vector_core::serde::is_default"
+    )]
+    pub timeout_millis: u64,
```

Comment:
Do you prefer to use `timeout_millis: u64` or `timeout_secs: f64`? How is usually done in this kind of config?

Reply:
We generally prefer `timeout_secs` with `f64`.